snap/README.md

```diff
@@ -0,0 +1,149 @@
+### Autostart
+By default, the edgex-device-grove disables its service on install, 
+as the expectation is that the default profile configuration files will be customized, 
+and thus this behavior allows the profile `configuration.toml` files in [`$SNAP_DATA`](https://snapcraft.io/docs/environment-variables) 
+to be modified before the service is first started.
+
+This behavior can be overridden by setting the `autostart` configuration setting to `true`. 
+This is useful when configuration and/or device profiles are being provided via configuration or gadget snap content interface.
+
+**Note** - this option is typically set from a gadget snap.
+
+### Rich Configuration
+While it's possible on Ubuntu Core to provide additional profiles via gadget 
+snap content interface, quite often only minor changes to existing profiles are required. 
+
+These changes can be accomplished via support for EdgeX environment variable 
+configuration overrides via the snap's configure hook.
+If the service has already been started, setting one of these overrides currently requires the
+service to be restarted via the command-line or snapd's REST API. 
+If the overrides are provided via the snap configuration defaults capability of a gadget snap, 
+the overrides will be picked up when the services are first started.
+
+The following syntax is used to specify service-specific configuration overrides:
+
+```
+env.<stanza>.<config option>
+```
+For instance, to setup an override of the service's Port use:
+```bash
+sudo snap set edgex-device-grove env.service.port=2112
+```
+And restart the service:
+```bash
+sudo snap restart edgex-device-grove.device-grove
+```
+
+**Note** - at this time changes to configuration values in the [Writable] section are not supported.
+For details on the mapping of configuration options to Config options, please refer to "Service Environment Configuration Overrides".
+
+## Service Environment Configuration Overrides
+**Note** - all of the configuration options below must be specified with the prefix: 'env.'
+```
+[Service]
+service.health-check-interval   // Service.HealthCheckInterval
+service.host                    // Service.Host
+service.server-bind-addr        // Service.ServerBindAddr
+service.port                    // Service.Port
+service.max-result-count        // Service.MaxResultCount
+service.max-request-size        // Service.MaxRequestSize
+service.startup-msg             // Service.StartupMsg
+service.request-timeout         // Service.RequestTimeout
+
+[SecretStore]
+secret-store.secrets-file               // SecretStore.SecretsFile
+secret-store.disable-scrub-secrets-file // SecretStore.DisableScrubSecretsFile
+
+[Clients.core-data]
+clients.core-data.port          // Clients.core-data.Port
+
+[Clients.core-metadata]
+clients.core-metadata.port      // Clients.core-metadata.Port
+
+[Device]
+device.update-last-connected    // Device.UpdateLastConnected
+device.use-message-bus          // Device.UseMessageBus
+```
```

Comment:
These are not implemented for this service. 

Instead explain where the configurations are (typically under `/var/snap/edgex-device-grove/current/config/edgex-device-grove/res`) and if a service restart is sufficient to apply the changes. Note that this service is started automatically, unlike in other device snaps.

Reply:
Removed and updated!